src/reducers/FormReducer.js

```diff
@@ -22,9 +22,13 @@ const initialState = config => {
     }),
     {}
   );
+  const formState = { isConfirmFormOpen: false };
+
+  return { formConfig, formState };
```

Comment:
Yeah, initially I didn't use it, but I think it's a bit tricky/cludgey not to as it breaks a lot of logic which maps/iterates over all the fields to display the form/check if it is valid etc.  

I think it is relatively nice to separate out the input configs from the other form state, but right now it's a bit stupid looking as there's only one field. I do think it'll come in handy though.

I agree re. the nesting, but I think on balance its worth it to prevent workarounds throughout the rest of the code. Also could potentially tidy this up using `combineReducers`!

Reply:
Sorry I just don't know how having `{formConfig: { .. }, isConfirmModalOpen: true }` really messes with the iteration as you're always iterating over `formConfig`? Any additional fields which would then need to be added to `formState` object would then comfortably sit in the first level and be easier to access so they're seperate and easier with less nesting!

Reply:
Sorry, I misunderstood (I thought you were suggesting having all fields as flat). Agree, just having them flat is good idea (although I still think there's something to be said for separating them and using `combineReducers`).